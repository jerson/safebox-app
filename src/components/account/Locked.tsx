import React from 'react';
import {
  StyleSheet,
  ViewStyle,
  TextStyle,
  View,
  ImageStyle,
  Image,
  Modal,
  StatusBar,
  ScrollView
} from 'react-native';
import Colors from '../../modules/constants/Colors';
import Button from '../ui/Button';
import TextInput from '../ui/TextInput';
import Text from '../ui/Text';
import useTextInput from '../hooks/useTextInput';
const tinyColor = require('tinycolor2');

const styles = StyleSheet.create({
  shadow: {
    shadowOpacity: 0.1,
    shadowRadius: 16,
    shadowColor: Colors.grey6,
    shadowOffset: { height: 4, width: 0 },
    elevation: 4
  } as ViewStyle,
  container: {
    padding: 20,
    margin: 20,
    maxWidth: 280,
    backgroundColor: Colors.white,
    borderRadius: 12,
    overflow: 'visible'
  } as ViewStyle,
  textInput: {
    marginTop: 0,
    borderColor: Colors.primaryLight
  } as ViewStyle,
  button: {} as ViewStyle,
  description: {
    fontSize: 13,
    color: Colors.grey6,
    textAlign: 'justify'
  } as TextStyle,
  help: {
    marginTop: 5,
    fontSize: 12,
    color: Colors.grey5
  } as TextStyle,
  logo: {
    width: 70,
    height: 70
  } as ImageStyle,
  logoContent: {
    backgroundColor: Colors.grey6,
    borderRadius: 15,
    borderWidth: 1,
    borderColor: Colors.accent,
    alignSelf: 'center',
    overflow: 'hidden',
    marginBottom: 10
  } as ViewStyle,
  scrollViewContent: {
    paddingBottom: 40,
    paddingTop: 20,
    alignItems: 'center',
    justifyContent: 'center'
  } as ViewStyle,
  scrollView: {
    backgroundColor: tinyColor(Colors.accentDark)
      .setAlpha(0.8)
      .toRgbString(),
    flex: 1
  }
});

export interface LockedProps {
  onUnlock: (password: string) => void;
  visible: boolean;
}

function Locked({ visible, onUnlock }: LockedProps) {
  const [password, passwordProps] = useTextInput('');
  const onUnlockCallback = () => {
    onUnlock(password);
  };

  return (
    <Modal transparent animated animationType={'fade'} visible={visible}>
      <StatusBar
        animated
        barStyle={'dark-content'}
        backgroundColor={Colors.grey5}
      />
      <ScrollView
        style={styles.scrollView}
        keyboardShouldPersistTaps={'handled'}
        contentContainerStyle={styles.scrollViewContent}
      >
        <View style={[styles.container, styles.shadow]}>
          <View style={styles.logoContent}>
            <Image
              style={styles.logo}
              resizeMode={'stretch'}
              source={require('../../assets/images/logo.png')}
            />
          </View>
          <Text style={styles.description}>
            To continue you must enter the password with which you Sign In to
            <Text style={{ color: Colors.primaryLight }}>
              {' '}
              Safe<Text weight={'Bold'}>Box</Text>
            </Text>
            .
          </Text>
          <TextInput
            icon={'lock'}
            secureTextEntry
            keyboardType={'default'}
            autoCapitalize={'none'}
            autoCorrect={false}
            autoCompleteType={'password'}
            returnKeyType={'done'}
            blurOnSubmit
            autoFocus
            onSubmitEditing={onUnlockCallback}
            style={styles.textInput}
            placeholder={'SafeBox password'}
            help={
              <Text style={styles.help}>
                SafeBox password is used to unlock your secret passwords.
              </Text>
            }
            {...passwordProps}
          />
          <Button
            icon={'unlock'}
            style={styles.button}
            onPress={onUnlockCallback}
            typeColor={'primaryLight'}
            title={'Unlock passwords'}
          />
        </View>
      </ScrollView>
    </Modal>
  );
}

export default Locked;
