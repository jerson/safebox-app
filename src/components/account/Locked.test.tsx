import React from 'react';
import * as renderer from 'react-test-renderer';
import Locked, { LockedProps } from './Locked';

const props: LockedProps = {
  onUnlock: jest.fn()
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<Locked {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<Locked {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
