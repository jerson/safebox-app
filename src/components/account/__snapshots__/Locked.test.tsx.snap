// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<Modal
  animated={true}
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 20,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "backgroundColor": "rgba(103, 129, 166, 0.8)",
        "flex": 1,
      }
    }
  >
    <_class>
      <_class
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "margin": 20,
              "maxWidth": 280,
              "overflow": "visible",
              "padding": 20,
            },
            Object {
              "elevation": 4,
              "shadowColor": "#38414e",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 16,
            },
          ]
        }
      >
        <_class
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#38414e",
              "borderColor": "#c2d0e7",
              "borderRadius": 15,
              "borderWidth": 1,
              "marginBottom": 10,
              "overflow": "hidden",
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "testUri": "../../../src/assets/images/logo.png",
              }
            }
            style={
              Object {
                "height": 70,
                "width": 70,
              }
            }
          />
        </_class>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Nunito",
                "fontSize": undefined,
                "fontStyle": "normal",
                "fontWeight": "400",
              },
              Object {},
              Object {
                "color": "#38414e",
                "fontSize": 13,
                "textAlign": "justify",
              },
            ]
          }
        >
          To continue you must enter the password with which you Sign In to
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Nunito",
                  "fontSize": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
                Object {},
                Object {
                  "color": "#9066f5",
                },
              ]
            }
          >
             
            Safe
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Nunito",
                    "fontSize": undefined,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                  },
                  Object {},
                  undefined,
                ]
              }
            >
              Box
            </Text>
          </Text>
          .
        </Text>
        <_class
          style={
            Array [
              Object {
                "flexDirection": "column",
                "marginBottom": 20,
                "marginTop": 10,
                "position": "relative",
              },
              undefined,
            ]
          }
        >
          <_class
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingBottom": 4,
                },
                undefined,
              ]
            }
          />
          <_class
            style={
              Array [
                Object {
                  "position": "relative",
                },
                undefined,
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCompleteType="password"
              autoCorrect={false}
              autoFocus={true}
              blurOnSubmit={true}
              editable={true}
              keyboardType="default"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="SafeBox password"
              placeholderTextColor="#65748b"
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={true}
              selectionColor="#65748b"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#f2f4f9",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "color": "#38414e",
                    "fontFamily": "Nunito",
                    "fontSize": 15,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "textDecorationLine": "none",
                  },
                  Object {
                    "paddingLeft": 40,
                  },
                  false,
                  false,
                  Object {
                    "paddingRight": 40,
                  },
                  Object {
                    "borderColor": "#9066f5",
                    "marginTop": 0,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <_class
              style={
                Object {
                  "bottom": 1,
                  "justifyContent": "center",
                  "left": 12,
                  "position": "absolute",
                  "top": 0,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#65748b",
                        "fontSize": 20,
                      },
                      false,
                      false,
                      undefined,
                    ],
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </_class>
            <_class
              style={
                Array [
                  Object {
                    "position": "absolute",
                    "right": -60,
                  },
                  undefined,
                ]
              }
            />
          </_class>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Nunito",
                  "fontSize": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
                Object {},
                Object {
                  "color": "#65748b",
                  "fontSize": 12,
                  "marginTop": 5,
                },
              ]
            }
          >
            SafeBox password is used to unlock your secret passwords.
          </Text>
        </_class>
        <_class
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#9066f5",
              "borderColor": "#9066f5",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 0,
              "opacity": 1,
              "padding": 12,
              "zIndex": 1,
            }
          }
          testID="button_touchable"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "marginRight": 4,
                    "top": 3,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Nunito",
                  "fontSize": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
                Object {},
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  undefined,
                ],
              ]
            }
          >
            Unlock passwords
          </Text>
        </_class>
      </_class>
    </_class>
  </RCTScrollView>
</Modal>
`;
